pkg/dns/dns.go

```diff
@@ -77,11 +93,12 @@ func NewKubeDNS(client *kclient.Client, domain string) *KubeDNS {
 }
 
 func (kd *KubeDNS) Start() {
-	go kd.eController.Run(wait.NeverStop)
+	go kd.endpointsController.Run(wait.NeverStop)
 	go kd.serviceController.Run(wait.NeverStop)
 	// Wait synchronously for the Kubernetes service and add a DNS record for it.
-	// TODO (abshah) UNCOMMENT AFTER TEST COMPLETE
-	//kd.waitForKubernetesService()
+	// This ensures that the Start function returns only after having received Service objects
+	// from APIServer.
+	kd.waitForKubernetesService()
```

Comment:
Does this NEED the master `kubernetes` service or would any service serve as proof-of-readiness?  Don't change behavior now, but maybe "TODO: better if this were not hard-coded"